includes/payment-methods/class-upe-split-payment-gateway.php

```diff
@@ -378,26 +378,32 @@ public function get_selected_stripe_payment_type_id() {
 	 *
 	 * @param string $intent_id     The payment intent id.
 	 * @param string $client_secret The payment intent client secret.
+	 *
+	 * @return void
 	 */
 	private function add_upe_payment_intent_to_session( string $intent_id = '', string $client_secret = '' ) {
-		$cart_hash = 'undefined';
+		$woocommerce = WC();
+		if ( isset( $woocommerce->session ) ) {
+			$cart_hash = 'undefined';
+			if ( isset( $_COOKIE['woocommerce_cart_hash'] ) ) {
+				$cart_hash = sanitize_text_field( wp_unslash( $_COOKIE['woocommerce_cart_hash'] ) );
+			}
```

Comment:
Not directly related to this PR, but I think it would be better to use the value from `WC()->cart`: 


```suggestion
			$cart_hash = isset( $woocommerce->cart ) ? $woocommerce->cart->get_cart_hash() : 'undefined';

```

Reply:
Nice! Addressed in 0b19357da700f17bbad9bb204cb5cf5a65d992f2.